[no-land] try fix 2

smessmer opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #26482 [wip] Error checking for KernelFunction::callUnboxed()
* #30712 Better error messages thanks to compile time type names
* **#31307 [no-land] try fix 2**
* #31306 [no-land] try fix 1
* #26628 Remove CAFFE_KNOWN_TYPE

Differential Revision: [D19085562](https://our.internmc.facebook.com/intern/diff/D19085562/)

smessmer committed:
[no-land] try fix 2



Differential Revision: [D19085562](https://our.internmc.facebook.com/intern/diff/D19085562/)

[ghstack-poisoned]

kostmo commented:
## CircleCI build failures summary
As of commit 593d303:
* **4/4** failures introduced in this PR



## Detailed failure analysis

One may explore the probable reasons each build failed interactively [on the Dr. CI website](https://dr.pytorch.org/commit-details.html?sha1=593d30309123e5fe4e9dd3da30c85d3e4e29b27d).

### 4 new failures recognized by patterns
The following build failures don't appear to be due to upstream breakage:
#### [![See CircleCI build](https://avatars0.githubusercontent.com/ml/7?s=12)](https://circleci.com/gh/pytorch/pytorch/3959641) binary_linux_libtorch_2_7m_cpu_gcc5_4_cxx11-abi_shared-with-deps_build (1/4)
**Step:** "Build" ([details](https://dr.pytorch.org/build-details.html?build_id=74829639))
<details>
<summary>
<code>Dec 16 06:28:33 FAILED: caffe2/CMakeFiles/caffe2_pybind11_state.dir/python/pybind_state_ideep.cc.o</code>
</summary>

```
Dec 16 06:28:31 ../c10/util/TypeIndex.h:25:22: note:   conversion of argument 1 would be ill-formed: 
Dec 16 06:28:31 ../c10/util/TypeIndex.h:24:8: note: candidate: constexpr c10::util::type_index::type_index(const c10::util::type_index&) 
Dec 16 06:28:31  struct type_index final : IdWrapper<type_index, uint64_t> { 
Dec 16 06:28:31         ^ 
Dec 16 06:28:31 ../c10/util/TypeIndex.h:24:8: note:   conversion of argument 1 would be ill-formed: 
Dec 16 06:28:31 ../c10/util/TypeIndex.h:24:8: note: candidate: constexpr c10::util::type_index::type_index(c10::util::type_index&&) 
Dec 16 06:28:31 ../c10/util/TypeIndex.h:24:8: note:   conversion of argument 1 would be ill-formed: 
Dec 16 06:28:32 [1555/2999] Building CXX object caffe2/CMakeFiles/TensorImpl_test.dir/__/aten/src/ATen/core/TensorImpl_test.cpp.o 
Dec 16 06:28:32 [1556/2999] Building CXX object caffe2/CMakeFiles/KernelFunction_test.dir/__/aten/src/ATen/core/boxing/KernelFunction_test.cpp.o 
Dec 16 06:28:33 [1557/2999] Building CXX object caffe2/CMakeFiles/caffe2_pybind11_state.dir/python/pybind_state_ideep.cc.o 
Dec 16 06:28:33 FAILED: caffe2/CMakeFiles/caffe2_pybind11_state.dir/python/pybind_state_ideep.cc.o  
E_GLOO -DHAVE_GCC_GET_CPUID -DUSE_AVX -DUSE_AVX2 -DTH_HAVE_THREAD -Wall -Wextra -Wno-unused-parameter -Wno-missing-field-initializers -Wno-write-strings -Wno-unknown-pragmas -Wno-missing-braces -fopenmp -std=gnu++14 -MD -MT caffe2/CMakeFiles/caffe2_pybind11_state.dir/python/pybind_state_ideep.cc.o -MF caffe2/CMakeFiles/caffe2_pybind11_state.dir/python/pybind_state_ideep.cc.o.d -o caffe2/CMakeFiles/caffe2_pybind11_state.dir/python/pybind_state_ideep.cc.o -c ../caffe2/python/pybind_state_ideep.cc 
Dec 16 06:28:33 In file included from ../c10/util/typeid.h:27:0, 
Dec 16 06:28:33                  from ../caffe2/core/context_base.h:11, 
Dec 16 06:28:33                  from ../caffe2/core/context.h:10, 
Dec 16 06:28:33                  from ../caffe2/python/pybind_state.h:5, 
Dec 16 06:28:33                  from ../caffe2/python/pybind_state_ideep.cc:7: 
Dec 16 06:28:33 ../c10/util/TypeIndex.h: In instantiation of ‘constexpr c10::util::type_index c10::util::get_type_index() [with T = std::__cxx11::basic_string<char>]’: 
Dec 16 06:28:33 ../c10/util/typeid.h:64:55:   required from ‘static constexpr caffe2::TypeIdentifier caffe2::TypeIdentifier::Get() [with T = std::__cxx11::basic_string<char>]’ 
Dec 16 06:28:33 ../c10/util/typeid.h:282:62:   required from ‘caffe2::detail::TypeMetaData caffe2::detail::_makeTypeMetaDataInstance() [with T = std::__cxx11::basic_string<char>]’ 
Dec 16 06:28:33 ../c10/util/typeid.h:424:45:   required from ‘static caffe2::TypeMeta caffe2::TypeMeta::Make() [with T = std::__cxx11::basic_string<char>]’ 
```

</details>

#### [![See CircleCI build](https://avatars0.githubusercontent.com/ml/7?s=12)](https://circleci.com/gh/pytorch/pytorch/3959645) binary_linux_manywheel_2_7mu_cpu_devtoolset7_build (2/4)
**Step:** "Build" ([details](https://dr.pytorch.org/build-details.html?build_id=74829808))
<details>
<summary>
<code>Dec 16 06:26:56 /pytorch/c10/util/typeid.h:282:31: error: constexpr call flows off the end of the function</code>
</summary>

```
Dec 16 06:26:56 /pytorch/c10/util/typeid.h:424:45:   required from ‘static caffe2::TypeMeta caffe2::TypeMeta::Make() [with T = float]’ 
Dec 16 06:26:56 /pytorch/c10/core/DefaultDtype.cpp:5:59:   required from here 
Dec 16 06:26:56 /pytorch/c10/util/typeid.h:72:22: note:   initializing argument 1 of ‘constexpr caffe2::TypeIdentifier::TypeIdentifier(c10::util::type_index)’ 
Dec 16 06:26:56    constexpr explicit TypeIdentifier(c10::util::type_index id) : IdWrapper(id) {} 
Dec 16 06:26:56                       ^~~~~~~~~~~~~~ 
Dec 16 06:26:56 /pytorch/c10/util/typeid.h: In instantiation of ‘caffe2::detail::TypeMetaData caffe2::detail::_makeTypeMetaDataInstance() [with T = float]’: 
Dec 16 06:26:56 /pytorch/c10/util/typeid.h:424:45:   required from ‘static caffe2::TypeMeta caffe2::TypeMeta::Make() [with T = float]’ 
Dec 16 06:26:56 /pytorch/c10/core/DefaultDtype.cpp:5:59:   required from here 
Dec 16 06:26:56 /pytorch/c10/util/typeid.h:282:62:   in constexpr expansion of ‘caffe2::TypeIdentifier::Get<float>()’ 
Dec 16 06:26:56 /pytorch/c10/util/typeid.h:64:55:   in constexpr expansion of ‘c10::util::get_type_index<float>()’ 
Dec 16 06:26:56 /pytorch/c10/util/typeid.h:282:31: error: constexpr call flows off the end of the function 
Dec 16 06:26:56    C10_HOST_CONSTEXPR_VAR auto typeId = TypeIdentifier::Get<T>(); 
Dec 16 06:26:56                                ^~~~~~ 
Dec 16 06:26:56 [  3%] Generating perfkernels/hp_emblookup_codegen.py 
Dec 16 06:26:56 gmake[2]: *** [c10/CMakeFiles/c10.dir/core/DefaultDtype.cpp.o] Error 1 
Dec 16 06:26:56 gmake[1]: *** [c10/CMakeFiles/c10.dir/all] Error 2 
Dec 16 06:26:56 gmake[1]: *** Waiting for unfinished jobs.... 
Dec 16 06:26:56 [  3%] Generating proto/__init__.py 
Dec 16 06:26:56 [  3%] Generating python/__init__.py 
Dec 16 06:26:56 [  3%] Building CXX object third_party/fbgemm/asmjit/CMakeFiles/asmjit.dir/src/asmjit/core/jitruntime.cpp.o 
Dec 16 06:26:56 [  3%] Generating python/_import_c_extension.py 
```

</details>

#### [![See CircleCI build](https://avatars0.githubusercontent.com/ml/7?s=12)](https://circleci.com/gh/pytorch/pytorch/3959652) pytorch_linux_xenial_py2_7_9_build (3/4)
**Step:** "Build" ([details](https://dr.pytorch.org/build-details.html?build_id=74829892))
<details>
<summary>
<code>Dec 16 06:29:23 /var/lib/jenkins/workspace/c10/util/typeid.h:282:31: error: constexpr call flows off the end of the function</code>
</summary>

```
Dec 16 06:29:23 /var/lib/jenkins/workspace/c10/util/typeid.h:424:45:   required from 'static caffe2::TypeMeta caffe2::TypeMeta::Make() [with T = c10::BFloat16]' 
Dec 16 06:29:23 /var/lib/jenkins/workspace/c10/core/ScalarType.h:179:5:   required from here 
Dec 16 06:29:23 /var/lib/jenkins/workspace/c10/util/typeid.h:72:22: note:   initializing argument 1 of 'constexpr caffe2::TypeIdentifier::TypeIdentifier(c10::util::type_index)' 
Dec 16 06:29:23    constexpr explicit TypeIdentifier(c10::util::type_index id) : IdWrapper(id) {} 
Dec 16 06:29:23                       ^~~~~~~~~~~~~~ 
Dec 16 06:29:23 /var/lib/jenkins/workspace/c10/util/typeid.h: In instantiation of 'caffe2::detail::TypeMetaData caffe2::detail::_makeTypeMetaDataInstance() [with T = c10::BFloat16]': 
Dec 16 06:29:23 /var/lib/jenkins/workspace/c10/util/typeid.h:424:45:   required from 'static caffe2::TypeMeta caffe2::TypeMeta::Make() [with T = c10::BFloat16]' 
Dec 16 06:29:23 /var/lib/jenkins/workspace/c10/core/ScalarType.h:179:5:   required from here 
Dec 16 06:29:23 /var/lib/jenkins/workspace/c10/util/typeid.h:282:62:   in constexpr expansion of 'caffe2::TypeIdentifier::Get<c10::BFloat16>()' 
Dec 16 06:29:23 /var/lib/jenkins/workspace/c10/util/typeid.h:64:55:   in constexpr expansion of 'c10::util::get_type_index<c10::BFloat16>()' 
Dec 16 06:29:23 /var/lib/jenkins/workspace/c10/util/typeid.h:282:31: error: constexpr call flows off the end of the function 
Dec 16 06:29:23    C10_HOST_CONSTEXPR_VAR auto typeId = TypeIdentifier::Get<T>(); 
Dec 16 06:29:23                                ^~~~~~ 
Dec 16 06:29:23 make[2]: *** [c10/CMakeFiles/c10.dir/core/StorageImpl.cpp.o] Error 1 
Dec 16 06:29:23 c10/CMakeFiles/c10.dir/build.make:86: recipe for target 'c10/CMakeFiles/c10.dir/core/StorageImpl.cpp.o' failed 
Dec 16 06:29:23 CMakeFiles/Makefile2:2012: recipe for target 'c10/CMakeFiles/c10.dir/all' failed 
Dec 16 06:29:23 make[1]: *** [c10/CMakeFiles/c10.dir/all] Error 2 
Dec 16 06:29:23 make[1]: *** Waiting for unfinished jobs.... 
Dec 16 06:29:23 [  7%] Building CXX object third_party/ideep/mkl-dnn/src/CMakeFiles/mkldnn.dir/cpu/rnn/ref_postgemm_gru.cpp.o 
Dec 16 06:29:23 [  7%] Built target gloo 
Dec 16 06:29:23 [  8%] Building CXX object third_party/ideep/mkl-dnn/src/CMakeFiles/mkldnn.dir/cpu/rnn/rnn_utils.cpp.o 
```

</details>

#### [![See CircleCI build](https://avatars0.githubusercontent.com/ml/7?s=12)](https://circleci.com/gh/pytorch/pytorch/3959639) binary_linux_libtorch_2_7m_cpu_devtoolset7_shared-with-deps_build (4/4)
**Step:** "Build" ([details](https://dr.pytorch.org/build-details.html?build_id=74830059))
<details>
<summary>
<code>Dec 16 06:27:26 /pytorch/c10/util/typeid.h:282:31: error: constexpr call flows off the end of the function</code>
</summary>

```
Dec 16 06:27:26 /pytorch/c10/util/typeid.h:424:45:   required from ‘static caffe2::TypeMeta caffe2::TypeMeta::Make() [with T = float]’ 
Dec 16 06:27:26 /pytorch/c10/core/DefaultDtype.cpp:5:59:   required from here 
Dec 16 06:27:26 /pytorch/c10/util/typeid.h:72:22: note:   initializing argument 1 of ‘constexpr caffe2::TypeIdentifier::TypeIdentifier(c10::util::type_index)’ 
Dec 16 06:27:26    constexpr explicit TypeIdentifier(c10::util::type_index id) : IdWrapper(id) {} 
Dec 16 06:27:26                       ^~~~~~~~~~~~~~ 
Dec 16 06:27:26 /pytorch/c10/util/typeid.h: In instantiation of ‘caffe2::detail::TypeMetaData caffe2::detail::_makeTypeMetaDataInstance() [with T = float]’: 
Dec 16 06:27:26 /pytorch/c10/util/typeid.h:424:45:   required from ‘static caffe2::TypeMeta caffe2::TypeMeta::Make() [with T = float]’ 
Dec 16 06:27:26 /pytorch/c10/core/DefaultDtype.cpp:5:59:   required from here 
Dec 16 06:27:26 /pytorch/c10/util/typeid.h:282:62:   in constexpr expansion of ‘caffe2::TypeIdentifier::Get<float>()’ 
Dec 16 06:27:26 /pytorch/c10/util/typeid.h:64:55:   in constexpr expansion of ‘c10::util::get_type_index<float>()’ 
Dec 16 06:27:26 /pytorch/c10/util/typeid.h:282:31: error: constexpr call flows off the end of the function 
Dec 16 06:27:26    C10_HOST_CONSTEXPR_VAR auto typeId = TypeIdentifier::Get<T>(); 
Dec 16 06:27:26                                ^~~~~~ 
Dec 16 06:27:26 [  8%] Built target gloo 
Dec 16 06:27:26 [  8%] Building CXX object third_party/fbgemm/CMakeFiles/fbgemm_generic.dir/src/GenerateKernelU8S8S32ACC32.cc.o 
Dec 16 06:27:26 gmake[2]: *** [c10/CMakeFiles/c10.dir/core/DefaultDtype.cpp.o] Error 1 
Dec 16 06:27:26 gmake[1]: *** [c10/CMakeFiles/c10.dir/all] Error 2 
Dec 16 06:27:26 gmake[1]: *** Waiting for unfinished jobs.... 
Dec 16 06:27:26 [  8%] Building CXX object third_party/fbgemm/CMakeFiles/fbgemm_generic.dir/src/GenerateKernelU8S8S32ACC32Avx512.cc.o 
Dec 16 06:27:27 [  8%] Building CXX object third_party/protobuf/cmake/CMakeFiles/libprotobuf.dir/__/src/google/protobuf/descriptor_database.cc.o 
Dec 16 06:27:27 [  8%] Building CXX object third_party/protobuf/cmake/CMakeFiles/libprotobuf.dir/__/src/google/protobuf/duration.pb.cc.o 
```

</details>



---
This comment was automatically generated by [Dr. CI](https://github.com/kostmo/circleci-failure-tracker/tree/master/docs/from-pull-request-comment).
Follow [this link to opt-out](https://dr.pytorch.org/admin/comments-opt-out.html) of these comments for your Pull Requests.


Please report bugs/suggestions on the [GitHub issue tracker](https://github.com/kostmo/circleci-failure-tracker/issues).

smessmer committed:
Update on "[no-land] try fix 2"


Differential Revision: [D19085562](https://our.internmc.facebook.com/intern/diff/D19085562/)

[ghstack-poisoned]

smessmer committed:
Update on "[no-land] try fix 2"


Differential Revision: [D19085562](https://our.internmc.facebook.com/intern/diff/D19085562/)

[ghstack-poisoned]

smessmer committed:
Update on "[no-land] try fix 2"


Differential Revision: [D19085562](https://our.internmc.facebook.com/intern/diff/D19085562/)

[ghstack-poisoned]

smessmer committed:
Update on "[no-land] try fix 2"


Differential Revision: [D19085562](https://our.internmc.facebook.com/intern/diff/D19085562/)

[ghstack-poisoned]

smessmer committed:
Update on "[no-land] try fix 2"


Differential Revision: [D19085562](https://our.internmc.facebook.com/intern/diff/D19085562/)

[ghstack-poisoned]

smessmer committed:
Update on "[no-land] try fix 2"


Differential Revision: [D19085562](https://our.internmc.facebook.com/intern/diff/D19085562/)

[ghstack-poisoned]

smessmer committed:
Update on "[no-land] try fix 2"


Differential Revision: [D19085562](https://our.internmc.facebook.com/intern/diff/D19085562/)

[ghstack-poisoned]

smessmer committed:
Update on "[no-land] try fix 2"


Differential Revision: [D19085562](https://our.internmc.facebook.com/intern/diff/D19085562/)

[ghstack-poisoned]
